ci/build_cpp.sh

```diff
@@ -17,9 +15,24 @@ rapids-logger "Begin cpp build"
 
 sccache --zero-stats
 
-RAPIDS_PACKAGE_VERSION=$(rapids-generate-version) rapids-conda-retry build \
-  conda/recipes/libcuml
+RAPIDS_PACKAGE_VERSION=$(rapids-generate-version)
+export RAPIDS_PACKAGE_VERSION
+
+source rapids-rattler-channel-string
+
+# --no-build-id allows for caching with `sccache`
+# more info is available at
+# https://rattler.build/latest/tips_and_tricks/#using-sccache-or-ccache-with-rattler-build
+rattler-build build --recipe conda/recipes/libcuml \
+                    --experimental \
+                    --no-build-id \
+                    --channel-priority disabled \
+                    --output-dir "$RAPIDS_CONDA_BLD_OUTPUT_DIR" \
+                    "${RATTLER_CHANNELS[@]}"
 
 sccache --show-adv-stats
 
+# remove build_cache directory
```

Comment:
For my understanding... is this necessary because `rattler-build` puts some files into `$RAPIDS_CONDA_BLD_OUTPUT_DIR` and `rapids-upload-conda-to-s3` is doing something like trying to zip up the entire contents of that directory?

I think it'd be helpful to modify this comment to answer the question "why?" instead of "what?"... it's clear to me from the code that this is removing a directory called `build_cache`, but not clear why.

Reply:
Yeah, there's an upstream issue tracking this that I can also link in there.